### ISSUE TITLE
SCSS If Else block not triggering issue with block-closing-brace-newline-before

### ISSUE THREAD

**Issue by lordmcfuzz** [stylelint-stylistic/stylelint-stylistic#58]
### What minimal example or steps are needed to reproduce the bug?

```SCSS
@mixin cuddle-test-bad($green: false) {
  @if $green {
    color: green;
  } @else {
    color: black;
  }
}

@mixin cuddle-test-good($green: false) {
  @if $green {
    color: green;
  } 
  @else {
    color: black;
  }
}
```

### What minimal configuration is needed to reproduce the bug?

```js
/** @type {import('stylelint').Config} */
export default {
    plugins: ["@stylistic/stylelint-plugin"],
    extends: ['stylelint-config-recommended-scss'],
    rules: {
      "@stylistic/block-closing-brace-newline-before": "always",
    }
  };
```

### How did you run Stylelint?

Package.json
```
"scripts": {
    "lint:scss": "stylelint \"src/test/*.scss\""
  }
```
Then run

```
npm run lint:scss
```

### Which Stylelint-related dependencies are you using?

```json
"devDependencies": {
    "@stylistic/stylelint-plugin": "^3.1.2",
    "stylelint": "^16.20.0",
    "stylelint-config-standard-scss": "^15.0.0"
  }
```

### What did you expect to happen?

I expect to see an error within `cuddle-test-bad` complaining that the `@else` needs to be on a new line.

`cuddle-test-good` is the expected formatting that I am trying to ensure.

### What actually happened?

No errors are found.

**Comment by lordmcfuzz** [comment]
@MorevM, you are very correct. I totally ran around in circles for hours, then distilling my entire setup for this report and didn't notice I was configuring the wrong rule. Thank you so much for this.
